### ISSUE TITLE
Master unit tests are failing after Espree upgrade

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#7169]
**Tell us about your environment**

* **ESLint Version:** Master branch
* **Node Version:** n/a
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?** Espree 3.2.0

After release of Espree 3.2.0 our unit tests started failing in master: See https://travis-ci.org/eslint/eslint/jobs/160511634 for example. This is not good, since we haven't enabled ES2017 anywhere yet, and all of the tests should continue to pass.

**Comment by platinumazure** [comment]
The Acorn change consumed by espree was semver-major. Maybe they introduced a general breaking change?

**Comment by not-an-aardvark** [comment]
The tests are failing due to a bugfix in Acorn. For example, this was previously parsed correctly by Acorn, but is actually invalid according to the ES2015 spec:

```js
function foo(bar, bar = 5) {
}
```

I think the test failures that we're seeing are cases where our tests used syntax that is invalid according to the spec.

**Comment by ilyavolodin** [comment]
I also just noticed that we don't get the name of the file that contains failing tests for some reason. I'm pretty sure we used to get that before.

**Comment by jordalgo** [comment]
+1 to this issue, seems like the upgrade to espree (3.2.0) caused it for me, because if I install 3.1.7 it doesn't happen.

**Comment by not-an-aardvark** [comment]
For the time being, would it be best to just remove the failing tests to fix the build? For reference, here are the tests that are failing:

```js
// no-dupe-args:
function a({a, b}, b) {}
function a([a, b], b) {}
function a([ , a], [b, , a]) {}
function a([a, b], {b}) {}

// no-redeclare:
export var a; export var a;
export class A {} export class A {}
```

All of them are SyntaxErrors; it looks like they were only working before because Acorn incorrectly parsed them rather than throwing an error.

**Comment by not-an-aardvark** [comment]
See #7170

**Comment by xjamundx** [comment]
There were also a few minor cases where the column and line numbers became more accurate.
